cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/concurrency/ExecutionConcurrencyManager.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2023 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.executor.concurrency;
+
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.ExecutorConfig;
+import com.linkedin.kafka.cruisecontrol.executor.ConcurrencyType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * ExecutionConcurrencyManager: manages the allowed concurrency for all execution types: inter-broker/intra-broker/leadership movement.
+ * For each execution type, the default allowed concurrency is defined by configuration, and the requested concurrency is an optional
+ *  parameter announced on triggering the rebalance. At the beginning of the execution, each broker starts with the same requested
+ *  concurrency (or the default concurrency if requested concurrency not provided). During the execution, each broker adjust its
+ *  allowed concurrency based on its own metric.
```

Comment:
This is useful, but it feels like there's enough to the concurrency adjustment that it deserves its own doc (or section of a doc) describing how it works and how it fits together with the other pieces. For example, how are recommendations acted upon, is concurrency ever changed by more than 1 in a single step, how quickly do recommendations get updated, is there hysteresis to prevent ping-ponging, etc. (I suspect you could hit the necessary points in one page or less.)

Such a document would be useful for onboarding new devs to CC and for reminding older or infrequent contributors of changes they may not recall.